scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -0,0 +1,1226 @@
+                     convert_to=str, default='docker')
+
+define_config_option( __monitor__, 'max_previous_lines',
+                     'Optional (defaults to 5000). The maximum number of lines to read backwards from the end of the stdout/stderr logs\n'
+                     'when starting to log a containers stdout/stderr to find the last line that was sent to Scalyr.',
+                     convert_to=int, default=5000)
+
+define_config_option( __monitor__, 'readback_buffer_size',
+                     'Optional (defaults to 5k). The maximum number of bytes to read backwards from the end of any log files on disk\n'
+                     'when starting to log a containers stdout/stderr.  This is used to find the most recent timestamp logged to file '
+                     'was sent to Scalyr.',
+                     convert_to=int, default=5*1024)
+
+define_config_option( __monitor__, 'log_mode',
+                     'Optional (defaults to "docker_api"). Determine which method is used to gather logs from the '
+                     'local containers. If "docker_api", then this agent will use the docker API to contact the local '
+                     'containers and pull logs from them.  If "syslog", then this agent expects the other containers '
+                     'to push logs to this one using the syslog Docker log plugin.  Currently, "syslog" is the '
+                     'preferred method due to bugs/issues found with the docker API.  It is not the default to protect '
+                     'legacy behavior.\n',
+                     convert_to=str, default="docker_api")
+
+define_config_option( __monitor__, 'metrics_only',
+                     'Optional (defaults to False). If true, the docker monitor will only log docker metrics and not any other information '
+                     'about running containers.\n',
+                     convert_to=bool, default=False)
+
+define_config_option( __monitor__, 'container_globs',
+                     'Optional (defaults to None). If true, a list of glob patterns for container names.  Only containers whose names '
+                     'match one of the glob patterns will be monitored.',
+                      default=None)
+
+define_config_option( __monitor__, 'report_container_metrics',
+                      'Optional (defaults to True). If true, metrics will be collected from the container and reported  '
+                      'to Scalyr.', convert_to=bool, default=True)
+
+# for now, always log timestamps to help prevent a race condition
+#define_config_option( __monitor__, 'log_timestamps',
+#                     'Optional (defaults to False). If true, stdout/stderr logs will contain docker timestamps at the beginning of the line\n',
+#                     convert_to=bool, default=False)
+
+define_metric( __monitor__, "docker.net.rx_bytes", "Total received bytes on the network interface", cumulative=True, unit="bytes", category="Network" )
```

Comment:
Is it still right to refer to this as docker metrics?  Do people running k8s think of it as also running docker?